src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1683,9 +1684,12 @@ class GracefulGoaway : public grpc_core::RefCounted<GracefulGoaway> {
         t, nullptr, GRPC_CLOSURE_INIT(&on_ping_ack_, OnPingAck, this, nullptr));
     grpc_chttp2_initiate_write(t, GRPC_CHTTP2_INITIATE_WRITE_GOAWAY_SENT);
     Ref().release();  // Ref for the timer
-    grpc_timer_init(
-        &timer_, grpc_core::Timestamp::Now() + grpc_core::Duration::Seconds(20),
-        GRPC_CLOSURE_INIT(&on_timer_, OnTimer, this, nullptr));
+    timer_handle_ =
+        t_->event_engine->RunAfter(grpc_core::Duration::Seconds(20), [this] {
+          grpc_core::ApplicationCallbackExecCtx callback_exec_ctx;
+          grpc_core::ExecCtx exec_ctx;
+          OnTimer();
```

Comment:
Since `OnTimer` is just one statement, and isn't called from anywhere else, you could move its body here and delete that function.

Reply:
Done.